components/aws/sagemaker/tests/unit_tests/tests/test_hpo.py

```diff
@@ -87,6 +87,23 @@ def test_main(self):
       call('/tmp/best_hyperparameters_output_path', {"key_1": "best_hp_1"}, json_encode=True),
       call('/tmp/training_image_output_path', 'training-image')
     ])
+
+  def test_main(self):
+    # Mock out all of utils except parser
+    hpo._utils = MagicMock()
+    hpo._utils.add_default_client_arguments = _utils.add_default_client_arguments
+
+    # Set some static returns
+    hpo._utils.create_hyperparameter_tuning_job.return_value = 'job-name'
```

Comment:
do we need return values for this test ?

Reply:
I can remove most of them. I have to keep `get_best_training_job_and_hyperparameters` because it needs to return a Tuple (that gets unpacked), rather than just `None`. I also have to keep `create_hyperparameter_tuning_job` because the `SIGTERM` test below relies on it. (This is a bug I have not investigated any further)